plugin-dev/Source/Sentry/Private/Android/SentrySpanAndroid.cpp

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) 2023 Sentry. All Rights Reserved.
+
+#include "SentrySpanAndroid.h"
+
+#include "Infrastructure/SentryConvertorsAndroid.h"
+#include "Infrastructure/SentryJavaClasses.h"
+
+SentrySpanAndroid::SentrySpanAndroid(jobject span)
+	: FSentryJavaObjectWrapper(SentryJavaClasses::Span, span)
+{
+	SetupClassMethods();
+}
+
+void SentrySpanAndroid::SetupClassMethods()
+{
+	FinishMethod = GetMethod("finish", "()V");
+	IsFinishedMethod = GetMethod("isFinished", "()Z");
+	SetTagMethod = GetMethod("setTag", "(Ljava/lang/String;Ljava/lang/String;)V");
+	SetDataMethod = GetMethod("setData", "(Ljava/lang/String;Ljava/lang/Object;)V");
+}
+
+void SentrySpanAndroid::Finish()
+{
+	CallMethod<void>(FinishMethod);
+}
+
+bool SentrySpanAndroid::IsFinished() const
+{
+	return false;
+}
+
+void SentrySpanAndroid::SetTag(const FString& key, const FString& value)
+{
+	CallMethod<void>(SetTagMethod, *GetJString(key), *GetJString(value));
+}
+
+void SentrySpanAndroid::RemoveTag(const FString& key)
+{
+	SetTag(key, TEXT(""));
```

Comment:
TIL.
Do you happen to know where this gets filtered? Is that tag still part of the event or does it get removed by the SDK?

Reply:
Based on a previous experience with setting extra context for captured events I assume that the "removed" tag is still sent to Sentry however it's not displayed due to its empty value. I wasn't able to find any code in sentry-java that removes it explicitly on the client side but it's hardly confirms my theory